lib/SQL/Abstract/mysql.pm

```diff
@@ -0,0 +1,263 @@
+        },
+        SCALARREF => sub { $group_sql = $$group }
+      }
+    );
+    $sql .= $self->_sqlcase(' group by ') . $group_sql;
+  }
+
+  # HAVING
+  if (defined(my $having = $options->{having})) {
+    my ($having_sql, @having_bind) = $self->_recurse_where($having);
+    $sql .= $self->_sqlcase(' having ') . $having_sql;
+    push @bind, @having_bind;
+  }
+
+  # ORDER BY
+  $sql .= $self->_order_by($options->{order_by}) if defined $options->{order_by};
+
+  # LIMIT
+  if (defined $options->{limit}) {
+    $sql .= $self->_sqlcase(' limit ') . '?';
+    push @bind, $options->{limit};
+  }
+
+  # OFFSET
+  if (defined $options->{offset}) {
+    $sql .= $self->_sqlcase(' offset ') . '?';
+    push @bind, $options->{offset};
+  }
+
+  # FOR
+  if (defined(my $for = $options->{for})) {
+    my $for_sql;
+    $self->_SWITCH_refkind(
+      $for => {
+        SCALAR => sub {
+          my %commands = (update => 'for update', share => 'lock in share mode');
+          my $command = $commands{$for} or puke qq{for value "$for" is not allowed};
+          $for_sql = $self->_sqlcase($command);
+        },
+        SCALARREF => sub { $for_sql = "FOR $$for" }
+      }
+    );
+    $sql .= " $for_sql";
+  }
+
+  return $sql, @bind;
+}
+1;
+
+=encoding utf8
+
+=head1 NAME
+
+SQL::Abstract::mysql - MySQL / MariaDB
+
+=head1 SYNOPSIS
+
+  use SQL::Abstract::mysql;
+  my $abstract = SQL::Abstract::mysql->new(quote_char => chr(96), name_sep => '.');
+  # The same as
+  use Mojo::mysql;
+  my $mysql = Mojo::mysql->new;
+  my $abstract = $mysql->abstract;
+
+  say $abstract->insert('some_table', \%some_values, \%some_options);
+  say $abstract->select('some_table');
+
+=head1 DESCRIPTION
+
+L<SQL::Abstract::mysql> extends L<SQL::Abstract> with a few MySQL / MariaDB features
+used by L<Mojo::mysql>. It was inspired by L<SQL::Abstract::Pg>.
+
+=head1 CONSTRUCTOR
+
+=head2 new
+
+  my $abstract = SQL::Abstract::mysql->new;
+
+=head1 METHODS
+
+L<SQL::Abstract::mysql> inherits all methods from L<SQL::Abstract>.
+
+=head2 insert
+
+  $abstract->insert($table, \@values || \%fieldvals, \%options);
+
+=head3 ON CONFLICT
+
+The C<on_conflict> option can be used to generate C<INSERT IGNORE>, C<REPLACE> and
+C<INSERT ... ON DUPLICATE KEY UPDATE> queries.
+So far C<'ignore'> to pass C<INSERT IGNORE>, C<'replace'> to pass C<REPLACE> and
+hash references to pass C<UPDATE> with conflict targets are supported.
+
+  # "insert ignore into t (id, a) values (123, 'b')"
+  $abstract->insert('t', {id => 123, a => 'b'}, {on_conflict => 'ignore'});
+
+  # "replace into t (id, a) values (123, 'b')"
+  $abstract->insert('t', {id => 123, a => 'b'}, {on_conflict => 'replace'});
+
+  # "insert into t (id, a) values (123, 'b') on duplicate key update c='d'"
+  $abstract->insert('t', {id => 123, a => 'b'}, {on_conflict => {c => 'd'}});
+
+=head2 select
+
+  $abstract->select($source, $fields, $where, $order);
+  $abstract->select($source, $fields, $where, \%options);
+
+=head3 AS
```

Comment:
@Tekki: I can't see where this is implemented. Was it planned, but then not finished..? I'm a bit surprised I didn't see this earlier...

Reply:
Yes, `sub _select_fields` is missing. I think the following happened: I started with the code from SQL::Abstract::Pg, then saw thought this method is not compatible with the JSON support I wanted to add, so I 'temporarily' removed it.
The question is what we do now, remove this paragraph from the docs or add the method?

Reply:
I would like to have this functionality :)

Reply:
Here is the PR: https://github.com/jhthorsen/mojo-mysql/pull/69